Does not seem to work with arm-none-eabi-binutils 2.39

Freshly cloned the repository, then:
```
$ git submodule update --init
$ cd examples/blue_pill_f103/blink
$ lbuild build
$ scons
```

Environment:
```
$ uname -a
Linux linas-pc 6.0.6-arch1-1 #1 SMP PREEMPT_DYNAMIC Sat, 29 Oct 2022 14:08:39 +0000 x86_64 GNU/Linux
```
```
$ arm-none-eabi-ld --version
GNU ld (GNU Binutils) 2.39
Copyright (C) 2022 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) a later version.
This program has absolutely no warranty.
```
```
$ arm-none-eabi-gcc --version
arm-none-eabi-gcc (Arch Repository) 12.2.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Build output:

```
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
Compiling C++·· release/main.o
Compiling C···· release/modm/ext/gcc/cabi.o
Compiling C++·· release/modm/ext/gcc/cxxabi.o
Compiling C++·· release/modm/ext/gcc/new_delete.o
Compiling C++·· release/modm/src/modm/architecture/driver/atomic/flag.o
Compiling C++·· release/modm/src/modm/board/board.o
Compiling C++·· release/modm/src/modm/math/utils/bit_operation.o
Compiling C++·· release/modm/src/modm/math/utils/pc/operator.o
Compiling C++·· release/modm/src/modm/platform/clock/rcc.o
Compiling C++·· release/modm/src/modm/platform/clock/systick_timer.o
Compiling C++·· release/modm/src/modm/platform/core/assert.o
Compiling C++·· release/modm/src/modm/platform/core/delay.o
Compiling C++·· release/modm/src/modm/platform/core/delay_ns.o
Compiling C···· release/modm/src/modm/platform/core/no_heap.o
Assembling····· release/modm/src/modm/platform/core/reset_handler.o
Compiling C···· release/modm/src/modm/platform/core/startup.o
Compiling C···· release/modm/src/modm/platform/core/startup_platform.o
Compiling C···· release/modm/src/modm/platform/core/vectors.o
Compiling C++·· release/modm/src/modm/platform/gpio/enable.o
Archiving······ release/modm/libmodm.a
Indexing······· release/modm/libmodm.a
Linking········ /home/linasdev/Projects/Embedded/asd/modm/build/blue_pill_f103/blink/scons-release/blink.elf
/usr/lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/bin/ld: warning: /home/linasdev/Projects/Embedded/asd/modm/build/blue_pill_f103/blink/scons-release/blink.elf has a LOAD segment with RWX permissions
collect2: error: ld returned 1 exit status
scons: *** [/home/linasdev/Projects/Embedded/asd/modm/build/blue_pill_f103/blink/scons-release/blink.elf] Error 1
scons: building terminated because of errors.
```

--- a/.github/workflows/macos.yml
+++ b/.github/workflows/macos.yml
@@ -24,7 +24,7 @@ jobs:
         run: |
           pip3 install --user modm scons
           echo "/usr/local/bin" >> $GITHUB_PATH
-          echo "/Users/runner/Library/Python/3.10/bin" >> $GITHUB_PATH
+          echo "/Users/runner/Library/Python/3.11/bin" >> $GITHUB_PATH
 
       - name: Dump environment
         run: |

--- a/src/modm/platform/core/cortex/hardware_init.hpp
+++ b/src/modm/platform/core/cortex/hardware_init.hpp
@@ -19,24 +19,24 @@
 /// @hideinitializer
 #define MODM_HARDWARE_INIT(function) \
 	modm_section(".hardware_init") \
-	void* MODM_CONCAT(__modm_hardware_init_ptr_, function) = (void*)&function
+	void* const MODM_CONCAT(__modm_hardware_init_ptr_, function) = (void*)&function
 
 /// Call `function` during boot process with a unique name.
 /// @hideinitializer
 #define MODM_HARDWARE_INIT_NAME(name, function) \
 	modm_section(".hardware_init") \
-	void* MODM_CONCAT(__modm_hardware_init_ptr_, name) = (void*)&function
+	void* const MODM_CONCAT(__modm_hardware_init_ptr_, name) = (void*)&function
 
 /// Call `function` during boot process in a global order.
 /// @hideinitializer
 #define MODM_HARDWARE_INIT_ORDER(function, order) \
 	modm_section(".hardware_init.order_" MODM_STRINGIFY(order)) \
-	void* MODM_CONCAT(__modm_hardware_init_ptr_, function) = (void*)&function
+	void* const MODM_CONCAT(__modm_hardware_init_ptr_, function) = (void*)&function
 
 /// Call `function` during boot process in a global order with a unique name.
 /// @hideinitializer
 #define MODM_HARDWARE_INIT_NAME_ORDER(name, function, order) \
 	modm_section(".hardware_init.order_" MODM_STRINGIFY(order)) \
-	void* MODM_CONCAT(__modm_hardware_init_ptr_, name) = (void*)&function
+	void* const MODM_CONCAT(__modm_hardware_init_ptr_, name) = (void*)&function
 
 /// @}

